### PR TITLE
Add Forget Password feature by sending email with Reset Password Link

### LINKED ISSUE #100: Feature: Reset Password with a verification email
* [x] Reset password modal for unauthenticated user
    * Add a modal with a `TextField` with `type="email"` and a submit button which will send an email containing the reset password link. Firebase SDK supports this feature and we can customize the email content on Firebase console. After submitting, close the modal and show a succes snackbar, ~~show a snackbar error if and only if a network issue occurs~~.
* [x] Reset password modal for unauthenticated user
   * Same as the modal for unauthenticated user and...
    * Fill the `TextField` with the current user's email
* [x] Change the "Need a new password?" text below to "Forget password?"

### PR DESCRIPTION
**Pull request by vitokhangnguyen**
Closes #100

### CODE REVIEW

File: src/modules/user-auth/components/ForgetPasswordModal/ForgetPasswordModal.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      size="md"
+      show={forgetPasswordModal.isShow}
+      onClose={
+        isAuthReady && !isAuthenticated
+          ? loginModal.show
+          : forgetPasswordModal.hide
+      }
+      header="Forget password?"
+      closeButtonIcon={
+        isAuthReady && !isAuthenticated ? (
```
Comment (Andrewnt219) [inline src/modules/user-auth/components/ForgetPasswordModal/ForgetPasswordModal.tsx:23]: Is `isAuthReady && !isAuthenticated` in both cases the same? Better store them as a variable with a suitable name.

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

**Comment by Andrewnt219** [comment]
Is this a spacer?

image

File: src/modules/user-auth/components/ForgetPasswordModal/ForgetPasswordModal.tsx (hunk 1 above)
Comment (vitokhangnguyen) [reply src/modules/user-auth/components/ForgetPasswordModal/ForgetPasswordModal.tsx:23]: On second though, `isAuthReady` is not necessary here... it is going to be false for a split second anyway

**Comment by vitokhangnguyen** [comment]
> Is this a spacer?
> 
> image

Yea, that space is to display error message when it appears. The dialog looks weird without it.

**Comment by Andrewnt219** [comment]
This isn't the icon used for error.

image

**Comment by vitokhangnguyen** [comment]
> This isn't the icon used for error.
> 

Copied it from LoginForm, apparently RegisterForm uses a different one. Fixed

**Review by Andrewnt219 — APPROVED** [review]

**Pull request merged**